DEVELOPING_OPAL/validate/src/test/java/org/opalj/fpcf/fixtures/immutability/openworld/assignability/clone_function/SimpleClonePattern.java

```diff
@@ -104,3 +104,17 @@ public CloneNonAssignableArrayWithRead clone(){
         return newInstance;
     }
 }
+
+class CloneAssignableArrayWithNonDominatedRead {
+
+    @AssignableField("field is assignable as one of the field reads is not dominated by a write")
+    private boolean[] booleans;
+
+    public CloneAssignableArrayWithNonDominatedRead clone(){
+        CloneAssignableArrayWithNonDominatedRead newInstance = new CloneAssignableArrayWithNonDominatedRead();
+        boolean nonDominated = newInstance.booleans[0];
+        newInstance.booleans = new boolean[this.booleans.length];
+        boolean dominated = newInstance.booleans[0];
+        return newInstance;
+    }
+}
```

Comment:
I think we should have at least the `nonDominated` escape in some way. While this field currently is recognized as assignable, one could reasonably argue that the dead read there could be recognized by a more sophisticated assignability analysis, or that an optimization pass before could get rid of it.

Reply:
I will add `System.out.println` to both test cases